packages/tendermint-rpc/src/adaptors/v0-33/responses.ts

```diff
@@ -405,22 +418,28 @@ type RpcSignature = {
   readonly signature: string;
 };
 
-function decodeSignature(data: RpcSignature): ValidatorSignature {
+function decodeCommitSignature(data: RpcSignature): CommitSignature {
   return {
-    algorithm: "ed25519",
-    data: fromBase64(assertNotEmpty(data.signature)),
+    blockIdFlag: decodeBlockIdFlag(data.block_id_flag),
+    validatorAddress: fromHex(data.validator_address),
+    timestamp: new Date(assertNotEmpty(data.timestamp)),
```

Comment:
Thanks for the Date and not ReadonlyDate here :)

Reply:
Using `Date` discards information. Do we need fractionl digits 4-9? Then we should create a `DateWithNanoseconds`, which works like `ReadonlyDateWithNanoseconds` but is a `Date` with an extra field that can be assigned to a `Date`.

Reply:
We don't need nanoseconds for now. We could create a new type as well.

Reply:
Let's make a new type when we need it.